src/accelerate/accelerator.py

```diff
@@ -1739,12 +1739,9 @@ def _prepare_megatron_lm(self, *args):
             else:
                 result.append(obj)
 
-        if model is not None:
-            model = megatron_lm_prepare_model(self)
-        if optimizer is not None:
-            optimizer = megatron_lm_prepare_optimizer(self, model)
-        if scheduler is not None:
-            scheduler = megatron_lm_prepare_scheduler(self, optimizer, scheduler)
+        model = megatron_lm_prepare_model(self)
+        optimizer = megatron_lm_prepare_optimizer(self, model)
+        scheduler = megatron_lm_prepare_scheduler(self, optimizer, scheduler)
```

Comment:
> What is the reason for this change in logic? I believe this should stay the same in case users don't pass in an opt or scheduler.
> 
> Otherwise cc @pacman100

Because if users don't pass in an opt or scheduler, it will use opt and scheduler to build `MegatronEngine`, so it must have opt and scheduler:
https://github.com/huggingface/accelerate/blame/8f9673f509c75defa4642826a3eafcd3f9735437/src/accelerate/accelerator.py#L1749-L1750
```python
if model is not None:
        model = MegatronEngine(self, model, optimizer, scheduler)
```